### PR TITLE
decompile script.s

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: src/script.c [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+#include "util.h"
+
+#define RAM_SCRIPT_MAGIC 51
+
+extern u8* gUnknown_020375C0;
+
+extern bool32 sub_801B27C(void);
+
+// ewram bss
+IWRAM_DATA u8 sScriptContext1Status;
```
Comment (camthesaxman) [inline src/script.c]: Make these actually static.

File: src/script.c [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    scriptData->magic = RAM_SCRIPT_MAGIC;
+    scriptData->mapGroup = mapGroup;
+    scriptData->mapNum = mapNum;
+    scriptData->objectId = objectId;
+    memcpy(scriptData->script, script, scriptSize);
+    gSaveBlock1Ptr->ramScript.checksum = CalculateRamScriptChecksum();
+    return TRUE;
+}
+
+u8 *GetRamScript(u8 objectId, u8 *script)
```
Comment (camthesaxman) [inline src/script.c]: Did this not work with the same if structure from pokeruby?

Comment (DizzyEggg) [reply src/script.c]: Nope, the ruby one is a 'fake-match'. :P

File: src/script.c (hunk 1 above)
Comment (DizzyEggg) [reply src/script.c]: Fixed.

File: src/script.c (hunk 2 above)
Comment (camthesaxman) [reply src/script.c]: Yeah, I tried porting the Ruby one (which looks better) here, and it had stupid register differences. Oh well.

**Pull request merged**